cmd/main_test.go

```diff
@@ -491,94 +507,58 @@ var _ = DescribeTable("TestGetNamespacesWithAccess", func(allNamespaces []k8sapi
+			allNamespaces = nil
 		})
 	})
 
-	Context("When only namespaces ns-test-tenant-1 and ns-test-tenant-2 has all necessary permissions and other's don't", func() {
-		mappings := []NamespaceRoleBinding{
-			{
-				Namespace:   "ns-test-tenant-1",
-				Role:        "ns-namespace-access-1",
-				RoleBinding: "ns-namespace-access-user-binding-1",
-			},
-			{
-				Namespace:   "ns-test-tenant-2",
-				Role:        "ns-namespace-access-2",
-				RoleBinding: "ns-namespace-access-user-binding-2",
-			},
-			{
-				Namespace:   "ns-test-tenant-3",
-				Role:        "ns-namespace-access-3",
-				RoleBinding: "ns-namespace-access-user-binding-3",
-			},
-			{
-				Namespace:   "ns-test-tenant-4",
-				Role:        "",
-				RoleBinding: "",
-			},
-		}
+	Context("When namspace ns-test-tenant-5 doesn't have necessary permissions", func() {
 		BeforeEach(func() {
-			for _, name := range mappings {
-				ns, err := createNamespace(k8sClient, name.Namespace)
-				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace %s: %v", name.Namespace, err))
+			var names = []string{"ns-test-tenant-5", "ns-test-tenant-6"}
+			for _, name := range names {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace %s", name))
 				allNamespaces = append(allNamespaces, ns)
+				expectedNs = []k8sapi.Namespace{ns}
 			}
-			createRole(k8sClient, "ns-test-tenant-1", "ns-namespace-access-1", []string{"create", "list", "watch", "delete"})
-			createRole(k8sClient, "ns-test-tenant-2", "ns-namespace-access-2", []string{"create", "list", "watch", "delete"})
-			createRole(k8sClient, "ns-test-tenant-3", "ns-namespace-access-3", []string{"create", "list", "watch"})
-			createRoleBinding(k8sClient, "ns-namespace-access-user-binding-1", "ns-test-tenant-1", "user1@konflux.dev", "ns-namespace-access-1")
-			createRoleBinding(k8sClient, "ns-namespace-access-user-binding-2", "ns-test-tenant-2", "user2@konflux.dev", "ns-namespace-access-2")
-			createRoleBinding(k8sClient, "ns-namespace-access-user-binding-3", "ns-test-tenant-3", "user3@konflux.dev", "ns-namespace-access-3")
-			expectedNs = append(expectedNs, allNamespaces[0], allNamespaces[1])
+			createRole(k8sClient, "ns-test-tenant-6", "ns-namespace-access-6", []string{"create", "list", "watch", "delete"})
+			createRoleBinding(k8sClient, "ns-namespace-access-user-binding-6", "ns-test-tenant-6", "user@konflux.dev", "ns-namespace-access-6")
 		})
-		It("returns only namespaces test-tenant and test-tenant-2", func() {
+		It("only returns ns-test-tenant-6 namespace", func() {
+			actualNs, err = getNamespacesWithAccess(e, c, authCl, allNamespaces)
 			Expect(actualNs).To(Equal(expectedNs))
 			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetNamespacesWithAccess")
 		})
 		AfterEach(func() {
-			deleteRoleBinding(k8sClient, "ns-test-tenant-1", "ns-namespace-access-user-binding-1")
```

Comment:
What surprising me is, didn't this all work when we just had the functional tests? I saw no issues with the removal...

Reply:
When I use the same name it was in terminating state and not getting deleted. When checked with @gbenhaim he mentioned we cannot delete the objects in [envtest environment](https://book.kubebuilder.io/reference/envtest#testing-considerations)